src/components/reports/ReportForm.tsx

```diff
@@ -0,0 +1,208 @@
+import React, {useState} from "react";
+import {Form, Field, Formik, ErrorMessage} from 'formik';
+import * as Yup from "yup"
+
+import Card from "../UI/Card";
+import classes from "./ReportForm.module.css";
+import {addressSchema, patientSchema, reportDetailSchema, sampleSchema, variantSchema} from "./formDataValidation";
+
+const FormValidation = Yup.object().shape({
+  address: addressSchema,
+  patient: patientSchema,
+  sample: sampleSchema,
+  variant: variantSchema,
+  result: reportDetailSchema,
+}).required();
+
+type FormValues = Yup.InferType<typeof FormValidation>;
+
+const initialValues: FormValues = {
+  address: {
+    streetAddress: [
+      "London North Genomic Laboratory Hub",
+      "Great Ormond Street Hospital for Children NHS Foundation Trust",
+      "Levels 4-6 Barclay House",
+      "37 Queen Square"],
+    city: "London",
+    country: "UK",
+    postCode: "WC1N 3BH",
+  },
+  patient: {
+    mrn: "",
+    firstName: "",
+    lastName: "",
+    dateOfBirth: "",
+    gender: "",
+    familyNumber: "",
+  },
+  sample: {
+    specimenCode: "",
+    specimenType: "",
+    collectionDate: "",
+    reasonForTestCode: "",
+    reasonForTestText: "",
+  },
+  variant: {
+    gene: "",
+    genomicHGVS: "",
+    inheritanceMethod: "",
+    classification: "",
+    proteinHGVS: "",
+    referenceNucleotide: "",
+    transcript: "",
+    variantNucleotide: "",
+    zygosity: "",
+    classificationEvidence: "",
+  },
+  result: {
+    resultSummary: "",
+    authorisingDate: "",
+    authorisingScientist: "",
+    authorisingScientistTitle: "",
+    furtherTesting: "",
+    reportingDate: "",
+    reportingScientist: "",
+    reportingScientistTitle: "",
+    testMethodology: "",
+  }
+};
+
+
+const ReportForm = () => {
+
+  const [result, setResult] = useState('')
+  return (
+    <Card>
+      <h1>Add a new report</h1>
+      <Formik
+        initialValues={initialValues}
+        validationSchema={FormValidation}
+        onSubmit={(values, actions) => {
+          setResult(JSON.stringify(values, null, 2));
+          actions.setSubmitting(false);
+        }}
+      >
```

Comment:
I've not used Formik before, but it looks decent aside from some potential cleanup of repetition.

Reply:
It seems quite nice because it's basically what I think I could do in terms of making my own components with some individual state each!